server/src/database/database-sqlite.js

```diff
@@ -0,0 +1,115 @@
+import sqlite3 from 'sqlite3'
+import path from 'path'
+import fs from 'fs'
+
+const sqlite = sqlite3.verbose()
+
+class SQLiteDatabase {
+  /*
+  Connects to database and tries to run new migrations
+   */
+  constructor(dbFile) {
+    this.databaseFile = dbFile || './dev_db.sqlite'
+    this.db = new sqlite.Database(this.databaseFile)
+  }
+
+  async initialize(dev) {
+    try {
+      const newestMigrationId = await this.query('SELECT ROWID FROM migrations ORDER BY ROWID DESC LIMIT 1')
+      const newestId = newestMigrationId[0].rowid
+      console.log('Connected to SQLite database at migration', newestId)
+      await this.runMigrations(newestId)
+    } catch (err) {
+      console.log('Could not load migrations, trying to add first migration')
+      await this.runMigrations(0)
+
+      if (dev) {
+        const sqlCommand = fs.readFileSync(path.resolve(__dirname, './test-data.sql')).toString()
+        await this.exec(sqlCommand)
+      }
+    }
+  }
+
+  /*
+  Gets all migrations from the folder ./migrations and runs all migrations
+  that are newer than the parameter newestId, then add them to migrations table in database
+   */
+
+  async runMigrations(newestId) {
+    const migrations = fs.readdirSync(path.resolve(__dirname, './migrations'))
+    migrations.sort()
+    for (const file of migrations) {
+      const migrNr = parseInt(file.slice(0, 3), 10)
+      if (migrNr > newestId) {
+        console.log(`Running migration ${file}`)
+        const sqlCommand = fs.readFileSync(path.resolve(__dirname, `./migrations/${file}`)).toString()
+        await this.exec(sqlCommand).then(() => {
+          this.run('INSERT INTO migrations(id, file) VALUES ((?), (?))', [migrNr, file]).catch((err) => {
+            console.error(`Failed to add ${file} to migrations table: ${err}`)
+          })
+        }, (err) => {
+          if (err) {
+            console.error(`Failed to run migration ${file}`, err)
+          }
+        })
+      }
+    }
+  }
+
+
+  /*
+  Runs an SQL query in the database. Does not return any value from the database,
+  only use for updating or inserting data
+
+  This only runs the first SQL query in the parameter string, to run many queries in the same
+  call , use exec instead
+   */
+
+  async run(...args) {
+    return new Promise((resolve, reject) => {
+      this.db.run(...args, (err) => {
+        if (err) {
+          console.error(`RUNERROR: ${err}`)
```

Comment:
Do we need these logs here?

Reply:
Should be rare so probably doesn't hurt

Reply:
Probably not needed now that the proper error handler is implemented, as it hsould take care of reporting errors?

Reply:
Probably doesn't hurt to keep it